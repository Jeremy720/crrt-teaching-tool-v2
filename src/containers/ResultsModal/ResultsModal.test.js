import React from "react";
import { ResultsModal, mapStateToProps } from "./";
import { shallow } from "enzyme";

describe("ResultsModal", () => {
  let wrapper;
  let mockTimestampArr = ["Hour 1", "Hour 2", "Hour 3"]

  beforeEach(() => {
    wrapper = shallow(<ResultsModal 
      hourlyTimestamps={mockTimestampArr} 
      caseOver={true} 
    />);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToPropsfunction", () => {
    it("should return an object with the timeStamps array", () => {
      const mockState = {
        hourlyTimestamps: mockTimestampArr,
        testProp: "testProp"
      };
  
      const expected = {
        hourlyTimestamps: mockTimestampArr
      };
  
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});