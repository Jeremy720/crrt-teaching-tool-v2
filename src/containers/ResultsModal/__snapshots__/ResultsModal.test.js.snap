// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsModal matches the snapshot 1`] = `undefined`;
